src/js/components/list.js

```diff
@@ -3,38 +3,18 @@ import AppStore from '../stores/app-stores';
 
 class List extends React.Component {
 
-  constructor(props){
-    super();
-    this.state = {currentList: AppStore.getList()};
-    this._onChange = this._onChange.bind(this);
-  }
-  componentWillMount(){
-    AppStore.addChangeListener( this._onChange )
-  }
-  componentWillUnmount(){
-    AppStore.removeChangeListener( this._onChange )
-  }
-  _onChange(){
-    this.setState( {currentList: AppStore.getList()} )
-  }
-
-
-  buildList(){
-    return this.state.currentList.map((item, i) => {
-      return (
-        <li key={i}>{item}</li>
-      )
-    })
-  }
-
   render(){
+    const { items } = this.props // destructure out from props
     return (
       <ul>
-        {this.buildList()}
+        {items.map(this.buildItem)}
```

Comment:
class javascript "this" issue here...

because the function "buildItem" is being passed directly, and executed by "map" then the "this" will be window

we can resolve in two ways:

1. Tell map to use the current this
``` items.map(this.buidItem, this)``` 

2. Bind the function explicitly
```items.map(this.buildItem.bind(this))```

3. Use a lambda
```items.map((item, i) => this.buildItem(item, i))```
but I dislike the repeating params

but here it doesn't matter since inside "buildItem" it's totally ignorant of this. I'll show in next commit how to do stateless components